libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2189,8 +2195,9 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 	else if (exprType->category() == Type::Category::Array)
 	{
 		auto const& arrayType(dynamic_cast<ArrayType const&>(*exprType));
+		bool isEmptyPush = memberName == "push" && arguments->numArguments() == 0;
 		annotation.isLValue = (
-			memberName == "length" &&
+			(memberName == "length" || isEmptyPush) &&
```

Comment:
Is this saying that `x.push` is an lvalue (as opposed to `x.push()`)?

Reply:
Removed that, should be taken care on the function call level.